trpc/trpc
# Internal types inference testing

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

I added an example project which should break if we don't export enough internal types from `@trpc/server`.

You can try playing with it by removing stuff from [internals.ts](https://github.com/trpc/trpc/blob/e1cff59b219605af47df557f4b005727c2168d22/packages/server/src/internals.ts) and see it breaking.

<del>

I used `pnpm` overrides to directly be linked against the live versions of `@trpc/client` and `@trpc/server` (useful trick found in [vitest](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md#testing-vitest-against-external-packages) repo!)

I'm not sure about the best way to run it in your CI workflow, basically, all you need is:

1. Install + build trpc monorepo packages
2. Go to `examples/.test/internal-types-inference`
3. Run `pnpm i`
4. Run `pnpm build`

If the `pnpm build` is successful, congrats, you broke nothing! 😄

</del>


**edit**: Please see https://github.com/trpc/trpc/pull/3789#issuecomment-1435105314

@KATT Would you mind finishing the work and integrating these commands into the current CI workflow?

I also audited each exported type from [internals.ts](https://github.com/trpc/trpc/blob/e1cff59b219605af47df557f4b005727c2168d22/packages/server/src/internals.ts) against both the example and my monorepo and I was able to delete a few of them. 👍

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.
